use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -296,33 +319,147 @@ main(int argc, char *argv[])
              l3_log_mode, loc_scheme);
 
 #else // L3_ENABLED
 
-    printf("Start Server, using clock ID=%d '%s': No logging.\n",
-           clock_id, clock_name(clock_id));
+    printf("Start Server, using clock ID=%d '%s': No logging"
+           ", %d server-threads, server-msgq-ID=%d.\n",
+           Clock_id, clock_name(Clock_id), num_threads, serverId);
     snprintf(run_descr, sizeof(run_descr), "Baseline - No logging");
 
 #endif // L3_ENABLED
 
-    /* Read requests, handle each in a separate child process */
-    Client_info *clientp = NULL;
-    requestMsg req;
-    responseMsg resp;
+    // Invoke pthread_create() to create n-threads ...
+    svr_thread_config   svr_config = {
+                            .server_id = serverId
+                        };
+    pthread_t   thread_ids[num_threads];
 
     struct timespec ts0 = {0};
-    struct timespec ts1 = {0};
 
+    if (svr_start_threads(thread_ids, num_threads, (void *) &svr_config)) {
+        errExit("Server thread creation failed.");
+    }
 #if !defined(__APPLE__)
-    if (clock_gettime(clock_id, &ts0)) {    // ***** Timing begins
+    if (clock_gettime(Clock_id, &ts0)) {        // ***** Timing begins
         errExit("clock_gettime-ts0");
     }
 #endif  // ! __APPLE__
 
+   // Wait for all threads to complete ...
+   for (int tctr = 0; tctr < num_threads; tctr++) {
+      void *thread_rc;
+      int   rc = pthread_join(thread_ids[tctr], &thread_rc);
+      if (rc) {
+        errExit("pthread_join() failed.");
+      }
+    }
+
+    struct timespec ts1 = {0};
+
+#if !defined(__APPLE__)
+    if (clock_gettime(Clock_id, &ts1)) {        // **** Timing ends
+        errExit("clock_gettime-ts1");
+    }
+#endif  // ! __APPLE__
+
+    assert(NumActiveClients == 0);
+
+    uint64_t nsec0 = timespec_to_ns(&ts0);
+    uint64_t nsec1 = timespec_to_ns(&ts1);
+    uint64_t elapsed_ns = (nsec1 - nsec0);
+
+    if (msgctl(serverId, IPC_RMID, NULL) == -1) {
+        errExit("msgctl serverId");
+    }
+
+    // Close L3-logging upon exit.
+#if defined(L3_LOGT_SPDLOG_BACKTRACE)
+
+    // Log the messages now! show the last n-messages
+    spdlog::dump_backtrace();
+
+#elif L3_ENABLED
+
+    l3_log_deinit(logtype);
+
+#endif  // L3_ENABLED
+
+    printf("Server: # active clients=%d (HWM=%d). Exiting.\n",
+           NumActiveClients, NumActiveClientsHWM);
+
+    printSummaryStats(outfile, run_descr, ActiveClients, NumActiveClientsHWM,
+                      Clock_id, elapsed_ns);
+
+    // For visibility into how clocks are performing on user's machine,
+    // run clock-calibration after all workload / metrics collection is done.
+
+#if defined(__cplusplus)
+
+#if !defined(L3_ENABLED) and !defined(L3_LOGT_SPDLOG) and !defined(L3_LOGT_SPDLOG_BACKTRACE)
+    svr_clock_calibrate();
+#endif
+
+#else   // __cplusplus
+
+#if !defined(L3_ENABLED)
+    svr_clock_calibrate();
+#endif  // L3_ENABLED
+
+#endif  // __cplusplus
+
+    exit(EXIT_SUCCESS);
+}
```

Comment:
No change from L362 thru here.

Just a diff-block that shows up due to code modularization and refactoring.